modxcms/revolution
# Add filtering for ACL policy and policy template grids

### What does it do?
This update adds a filtering toolbar and associated methods for the referenced grids. It also provides some housekeeping and enhancements:
- Removes some unneeded code, such as the `url` and `baseParams` config entries where they are not applicable (since these are local not remote grids)
- For the Create form, changed the `name` field's xtype to textfield and removed code associated with the former combobox (which wasn't working and was irrelevant in the context of creating a new entry).
- Added dirty marker to new entries, as new entries are only saved when the user clicks save, and added new row highlight give additional cue to user that their new data was added but needs further action (save).
- New entries now appear at the top of the grid instead of the bottom
- Grids re-sort on save so new entries are put in order
- Added ability to search both the `name` _and_ `description` (or just name via checkbox control in toolbar)

### Why is it needed?
Addresses a very old request for this functionality, which makes the longer policy/policy template grids (the administrator ones in particular) much easier to navigate and work with.

### How to test
1. Run `grunt build` from `_build/templates/default` and ensure your caches are fully cleared.
2. Go to both the Policies and Policy Templates areas, select a specific Policy / Policy Template (especially the admin ones) and experiment with the new search filter.
3. Make changes/additions/updates to each to verify expected functionality.

### Related issue(s)/PR(s)
Resolves #5822.
